torch/utils/_python_dispatch.py

```diff
@@ -87,17 +90,20 @@ def _push_mode(mode, k: Optional[DispatchKey] = None):
         _push_on_torch_dispatch_stack(mode)
 
 
-def _pop_mode(k: Optional[DispatchKey] = None):
+def _pop_mode(k: Optional[DispatchKey] = None, mode_key: Optional[torch._C.TorchDispatchModeKey] = None):
```

Comment:
This seems like a pretty weird to write the API. Isn't `k: Union[None, DispatchKey, ModeKey]` more appropriate?

Reply:
I think the union makes more sense today, yep - in particular, since the "DispatchKey" bit is only used in our per-dispatch-key-mode logic, and right now, all of the `TorchDispatchModeKey` is specific to the `Python` key.